### DISCUSSION TITLE
Exporting variables during solver execution

### DISCUSSION THREAD

**Discussion by gavilomon** [casadi/casadi#3520] [Q&A] [answered]
Hi!

I'm currently working on exporting intermediate solver (ipopt) parameters using the opti class in MATLAB for my optimal control code. Within a function, I aim to output values like the objective function's iteration-wise data, so I can smoothly plot in a MATLAB GUI. Additionally, I'm interested in variables such as velocity at each iteration.

Although I've managed to plot these values within my code using the opti.callback method, I'm facing difficulty in exporting this information from the opti class during solver execution and consequently from my function, so it can be used by my GUI.

Is there a method to achieve this?

Thanks in advance!

G

**Comment by jgillis** [comment, marked as answer]
I could imagine you could create a global workspace variable, initialized as a cell, to which you append those intermediate outputs.

**Reply by gavilomon** [reply]
Solved thanks!
